tests/API/TimesheetControllerTest.php

```diff
@@ -379,6 +379,45 @@ public function testInvalidPatchAction()
         $this->assertApiCallValidationError($response, ['end', 'activity']);
     }
 
+    public function testDeleteAction()
+    {
+        $client = $this->getClientForAuthenticatedUser(User::ROLE_USER);
+        // first insert an entry
+        $data = [
+            'activity' => 1,
+            'project' => 1,
+            'begin' => (new \DateTime('- 8 hours'))->format('Y-m-d H:m'),
+            'end' => (new \DateTime())->format('Y-m-d H:m'),
+            'description' => 'foo',
+            'fixedRate' => 2016,
+            'hourlyRate' => 127
+        ];
+        $this->request($client, '/api/timesheets', 'POST', [], json_encode($data));
+        $this->assertTrue($client->getResponse()->isSuccessful());
+
+        $result = json_decode($client->getResponse()->getContent(), true);
+        $this->assertIsArray($result);
+        $this->assertDefaultStructure($result);
+        $this->assertNotEmpty($result['id']);
+        $id = $result['id'];
+
+        // now delete it
+        $data = [
+            'activity' => 1,
+            'project' => 1,
+            'begin' => (new \DateTime('- 8 hours'))->format('Y-m-d H:m'),
+            'end' => (new \DateTime())->format('Y-m-d H:m'),
+            'description' => 'foo',
+            'fixedRate' => 2016,
+            'hourlyRate' => 127
+        ];
+        $this->request($client, "/api/timesheets/$id", 'DELETE', [], json_encode($data));
+        $this->assertTrue($client->getResponse()->isSuccessful());
+
+        // check whether id is removed
+        $this->assertAccessDenied($client, "/api/timesheets/$id");
```

Comment:
Please use `$this->assertEntityNotFound(User::ROLE_USER, sprintf('/api/timesheets/:s', $id));` instead